src/main/java/org/folio/inventory/dataimport/handlers/actions/HoldingsUpdateDelegate.java

```diff
@@ -0,0 +1,101 @@
+package org.folio.inventory.dataimport.handlers.actions;
+
+import static java.lang.String.format;
+
+import java.util.Map;
+
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.json.JsonObject;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import org.folio.Holdings;
+import org.folio.HoldingsRecord;
+import org.folio.inventory.common.Context;
+import org.folio.inventory.domain.HoldingsRecordCollection;
+import org.folio.inventory.storage.Storage;
+import org.folio.processing.mapping.defaultmapper.RecordMapper;
+import org.folio.processing.mapping.defaultmapper.RecordMapperBuilder;
+import org.folio.processing.mapping.defaultmapper.processor.parameters.MappingParameters;
+import org.folio.rest.jaxrs.model.ParsedRecord;
+import org.folio.rest.jaxrs.model.Record;
+
+public class HoldingsUpdateDelegate {
+
+  private static final Logger LOGGER = LogManager.getLogger(HoldingsUpdateDelegate.class);
+
+  private static final String MAPPING_RULES_KEY = "MAPPING_RULES";
+  private static final String MAPPING_PARAMS_KEY = "MAPPING_PARAMS";
+  private static final String MARC_FORMAT = "MARC_HOLDINGS";
+
+  private final Storage storage;
+
+  public HoldingsUpdateDelegate(Storage storage) {
+    this.storage = storage;
+  }
+
+  public Future<HoldingsRecord> handle(Map<String, String> eventPayload, Record marcRecord, Context context) {
+    try {
+      JsonObject mappingRules = new JsonObject(eventPayload.get(MAPPING_RULES_KEY));
+      MappingParameters mappingParameters =
+        new JsonObject(eventPayload.get(MAPPING_PARAMS_KEY)).mapTo(MappingParameters.class);
+
+      JsonObject parsedRecord = retrieveParsedContent(marcRecord.getParsedRecord());
+      String holdingsId = marcRecord.getExternalIdsHolder().getHoldingsId();
```

Comment:
Do we have such that the `getExternalIdsHolder` will be empty? Do we need to avoid NPE?

Reply:
If it is empty, there is no chance to somehow handle this event. It should be there in any case. So, there is no difference between throwing NPE there or checking ExternalIdsHolder for NULL and throwing some other exception. Maybe it will be more clear. Tell me if I should handle this NULL-check.